cloudant-client/src/test/java/com/cloudant/tests/CloudFoundryServiceTest.java

```diff
@@ -0,0 +1,170 @@
+            }
+        }
+
+        public void createNewService(String name, String url, String username, String password) {
+            addService(name, url, username, password);
+        }
+
+        public void createNewServiceWithEmptyCredentials(String name) {
+            addService(name, null, null, null);
+        }
+
+        public void createNewServiceWithoutName(String url, String username, String password) {
+            addService(null, url, username, password);
+        }
+
+        public String toJson() {
+            return new GsonBuilder().create().toJson(vcap);
+        }
+    }
+
+    @Test
+    public void vcapSingleService() {
+        new MockUp<System>() {
+            @Mock
+            public String getenv(final String string) {
+                VCAPGenerator vcap = new CloudFoundryServiceTest.VCAPGenerator();
+                vcap.createNewService("vcapSingleService",
+                        CloudantClientHelper.SERVER_URI_WITH_USER_INFO,
+                        CloudantClientHelper.COUCH_USERNAME, CloudantClientHelper.COUCH_PASSWORD);
+                return vcap.toJson();
+            }
+        };
+        ClientBuilder.bluemix("vcapSingleService").build().serverVersion();
+    }
+
+    @Test
+    public void vcapSingleServiceNoName() {
+        new MockUp<System>() {
+            @Mock
+            public String getenv(final String string) {
+                VCAPGenerator vcap = new CloudFoundryServiceTest.VCAPGenerator();
+                vcap.createNewServiceWithoutName(CloudantClientHelper.SERVER_URI_WITH_USER_INFO,
+                        CloudantClientHelper.COUCH_USERNAME, CloudantClientHelper.COUCH_PASSWORD);
+                return vcap.toJson();
+            }
+        };
+        ClientBuilder.bluemix().build().serverVersion();
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void vcapSingleServiceMissingService() {
+        new MockUp<System>() {
+            @Mock
+            public String getenv(final String string) {
+                return new CloudFoundryServiceTest.VCAPGenerator().toJson();
+            }
+        };
+        ClientBuilder.bluemix();
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void vcapSingleServiceMultipleServiceWithNoNameSpecified() {
+        new MockUp<System>() {
+            @Mock
+            public String getenv(final String string) {
+                VCAPGenerator vcap = new CloudFoundryServiceTest.VCAPGenerator();
+                vcap.createNewServiceWithoutName(CloudantClientHelper.SERVER_URI_WITH_USER_INFO,
+                        CloudantClientHelper.COUCH_USERNAME, CloudantClientHelper.COUCH_PASSWORD);
+                vcap.createNewServiceWithoutName(CloudantClientHelper.SERVER_URI_WITH_USER_INFO,
+                        CloudantClientHelper.COUCH_USERNAME, CloudantClientHelper.COUCH_PASSWORD);
+                return vcap.toJson();
+            }
+        };
+        ClientBuilder.bluemix();
+    }
+
```

Comment:
I think we could use another test for the other `IllegalArgumentException` case something like:
`vcapMultiServiceMissingNamedService`